outboarding_guide_outline.md

```diff
@@ -0,0 +1,109 @@
+# Onboarding Guide Outline
+
+## Overview
+
+Each major language that Tilt supports should have an onboarding guide that demonstrates:
+- The simplest way to set up Tilt with that language
+- The recommended way to set up Tilt with that language ("the fastest deployment time")
+- Walkthrough a clear path between those two setups
+
+This guide should have working code samples that the user can check out and explore themselves.
+
+We're basically cannibalizing the narrative of "The Quest for the Fastest Deployment Time"
+
+Slides: https://docs.google.com/presentation/d/1Vs4a4SSI4BE2094bloBY410FJkaH1WucHvVXcOJ78H8/edit#slide=id.g76703e551a_2_427
+Code: https://github.com/windmilleng/fast
+
+But with a couple changes in scope:
+
+- More detailed explanation of what the Tiltfile does
+- Fewer intermediate experiments
+- A simpler app
+
+## Code Structure
+
+All repos should borrow the code structure from https://github.com/windmilleng/fast
+
+The repo name should be `windmilleng/tilt-example-LANGUAGE`. So Go would be
+`tilt-example-go`, Java would be `tilt-example-java`, etc.
+
+Each subdirectory would be a version of the app:
+
+- 0-base: The simplest possible app
+- 1-measured: The simplest possible app, but modified to measure its own deployment time.
+- 2-name-a: First optimization
+- 3-name-b: And so on
+- 4-recommended: The recommended version of the app.
+
+## Narrative Structure
+
+The best indicator of a healthy development workflow is a short feedback loop.
+
+Kubernetes is a huge wrench in the works.
+
+Let’s fix this.
+
+### The Initial Setup
+
+A simple Hello World server.
+
+Running on Kubernetes
+
+Set up for Tilt
+
+Introduce Deployment yaml
+
+Introduce Dockerfile
+
+Introduce Tiltfile
+
+Link to snapshot of the server running
+
+### Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Add a local_resource that records the current time, and kicks off a build.
+
+Add code to the app that records start up time.
+
+Let's see it!
```

Comment:
Might be worth it to split this off into a generic blog post or short guide, and just link to it instead of writing it again every time.

Reply:
Ya, that's a good question!

There are two separable things I struggle with here: 
(1) if it's useful to introduce local_resource early in onboarding, and
(2) what the line should be between "writing a generic guide" vs "keeping them in the language-specific guide" flow

re: (1) I'd like to at least try introducing local_resource somewhere in this guide and and see how it goes. It's a great escape hatch for the person setting up Tilt and fitting it into arbitrary projects. It seems like something people ask for a lot

re: (2) When I look at docs for comparable tools, they try to have a self-contained language-specific guide, cf: 
https://circleci.com/docs/2.0/language-java/
https://circleci.com/docs/2.0/language-go/
each has their own code for setting up postgres, though I agree that not every aspect of Tilt should have a language-specific guide

Reply:
Alright, good points!